.github/workflows/wasmcloud.yml

```diff
@@ -242,159 +242,123 @@ jobs:
       id: deployment
 
   oci:
+    strategy:
+      matrix:
+        include:
+        - bin: wasmcloud
+
+        - bin: wash
+          prefix: wash-cli-
+
     runs-on: ubuntu-22.04
     permissions:
       packages: write
     needs:
     - build-bin
     - test-linux
-    if: ${{ github.event_name != 'pull_request' && github.event_name != 'merge_group' }}
-    env:
-      # Annotations cause issues with the image metadata
-      # as displayed on GitHub packages page for the repo ('unknown/unknown')
-      #
-      # See: https://github.com/orgs/community/discussions/45969
-      BUILDX_NO_DEFAULT_ATTESTATIONS: 1
     steps:
     - uses: actions/checkout@v4.1.1
     - uses: ./.github/actions/install-nix
-      if: startsWith(github.ref, 'refs/tags/v') || github.ref == 'refs/heads/main'
       with: 
         cachixAuthToken: '${{ secrets.CACHIX_AUTH_TOKEN }}'
 
     - name: Extract tag context
-      if: startsWith(github.ref, 'refs/tags/v') || github.ref == 'refs/heads/main'
       id: ctx
       run: |
-        echo 'package=${GITHUB_REPOSITORY,,}' >> $GITHUB_OUTPUT
-        echo 'sha_short=$(git rev-parse --short HEAD)' >> $GITHUB_OUTPUT
-        echo 'version=${GITHUB_REF_NAME#v}' >> $GITHUB_OUTPUT
+        echo "owner=${GITHUB_REPOSITORY_OWNER,,}" >> "$GITHUB_OUTPUT"
+        echo "sha_short=$(git rev-parse --short HEAD)" >> "$GITHUB_OUTPUT"
+
+        version=${GITHUB_REF_NAME#${{ matrix.prefix }}v}
+        echo "version=${version}" >> "$GITHUB_OUTPUT"
+        echo "version is ${version}"
+        if [[ $version == *"-"* ]]; then
+          echo "version ${version} is a pre-release"
+          echo "prerelease=true" >> "$GITHUB_OUTPUT"
+        fi
 
     - name: Login to AzureCR
-      if: startswith(github.ref, 'refs/tags/v') || startswith(github.ref, 'refs/tags/wash-cli-v')
+      if: startswith(github.ref, format('refs/tags/{0}v', matrix.prefix)) || github.ref == 'refs/heads/main'
+      continue-on-error: ${{ github.repository_owner != 'wasmCloud' }}
```

Comment:
Is this to handle forks? I'm not sure I understand

Reply:
Yes, see commit https://github.com/wasmCloud/wasmCloud/pull/953/commits/e579676fc40e018fb417caeadf831771ae48fdcd